test/saml-post-signing-tests.spec.ts

```diff
@@ -301,4 +301,75 @@ describe("SAML POST Signing", function () {
       },
     });
   });
+
+  it("should sign an AuthnRequest and include Signature KeyInfo", async function () {
+    const xml =
+      '<AuthnRequest xmlns="urn:oasis:names:tc:SAML:2.0:protocol"><saml2:Issuer xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion">http://example.com</saml2:Issuer></AuthnRequest>';
+    const signingCert = fs.readFileSync(__dirname + "/static/cert.pem").toString();
+    const result = signAuthnRequestPost(xml, { privateKey: signingKey, signingCert });
+    const doc = await parseXml2JsFromString(result);
+    doc.should.be.deepEqual({
+      AuthnRequest: {
+        $: { xmlns: "urn:oasis:names:tc:SAML:2.0:protocol", Id: "_0" },
+        Issuer: [
+          {
+            _: "http://example.com",
+            $: { "xmlns:saml2": "urn:oasis:names:tc:SAML:2.0:assertion" },
+          },
+        ],
+        Signature: [
+          {
+            $: { xmlns: "http://www.w3.org/2000/09/xmldsig#" },
+            SignedInfo: [
+              {
+                CanonicalizationMethod: [
+                  { $: { Algorithm: "http://www.w3.org/2001/10/xml-exc-c14n#" } },
+                ],
+                SignatureMethod: [
+                  { $: { Algorithm: "http://www.w3.org/2000/09/xmldsig#rsa-sha1" } },
+                ],
+                Reference: [
+                  {
+                    $: { URI: "#_0" },
+                    Transforms: [
+                      {
+                        Transform: [
+                          {
+                            $: {
+                              Algorithm: "http://www.w3.org/2000/09/xmldsig#enveloped-signature",
+                            },
+                          },
+                          { $: { Algorithm: "http://www.w3.org/2001/10/xml-exc-c14n#" } },
+                        ],
+                      },
+                    ],
+                    DigestMethod: [{ $: { Algorithm: "http://www.w3.org/2000/09/xmldsig#sha1" } }],
+                    DigestValue: [{ _: "wHDDyV7rEQ/AQLYeLgsEUXX+Zxw=" }],
+                  },
+                ],
+              },
+            ],
+            KeyInfo: [
+              {
+                X509Data: [
+                  {
+                    X509Certificate: [
```

Comment:
Instead of checking for deep equal, why not just check for the keys you want to be present? Otherwise, this tests ends up testing for more than it is designed to and may fail if other code adds more features or fixes unrelated bugs at another time.